eth/stagedsync/stage_headers.go

```diff
@@ -366,6 +372,15 @@ func startHandlingForkChoice(
 		}
 	}
 
+	if headerHash == cfg.hd.GetNextForkHash() && cfg.memoryOverlay {
```

Comment:
Just nit-picking, but it is better to put `cfg.memoryOverlay` first, because of the `&&` short-circuiting. In `A && B`, if `A` is false, then `B` is not even evaluated